qiskit_experiments/library/randomized_benchmarking/mirror_rb_analysis.py

```diff
@@ -0,0 +1,674 @@
+            hamming_dists = np.zeros(self._num_qubits + 1)
+            for bitstring, count in circ_result["counts"].items():
+                # Compute success probability
+                success_prob = 0.0
+                if bitstring == target_bs:
+                    success_prob = count / circ_result.get(
+                        "shots", sum(circ_result["counts"].values())
+                    )
+                    success_prob_unc = np.sqrt(success_prob * (1 - success_prob))
+                    if self.options.y_axis == "Success Probability":
+                        y_data.append(success_prob)
+                        y_data_unc.append(success_prob_unc)
+                    circ_result["metadata"]["success_probability"] = success_prob
+                    circ_result["metadata"]["success_probability_stddev"] = success_prob_unc
+
+                # Compute hamming distance proportions
+                target_bs_to_list = [int(char) for char in target_bs]
+                actual_bs_to_list = [int(char) for char in bitstring]
+                k = int(round(hamming(target_bs_to_list, actual_bs_to_list) * self._num_qubits))
+                hamming_dists[k] += count / circ_result.get(
+                    "shots", sum(circ_result["counts"].values())
+                )
+
+            # Compute hamming distance uncertainties
+            hamming_dist_unc = np.sqrt(hamming_dists * (1 - hamming_dists))
+
+            # Compute adjusted success probability and standard deviation
+            adjusted_success_prob = 0.0
+            adjusted_success_prob_unc = 0.0
+            for k in range(self._num_qubits + 1):
+                adjusted_success_prob += (-0.5) ** k * hamming_dists[k]
+                adjusted_success_prob_unc += (0.5) ** k * hamming_dist_unc[k] ** 2
+            adjusted_success_prob_unc = np.sqrt(adjusted_success_prob_unc)
+            circ_result["metadata"]["adjusted_success_probability"] = adjusted_success_prob
+            circ_result["metadata"][
+                "adjusted_success_probability_stddev"
+            ] = adjusted_success_prob_unc
+            if self.options.y_axis == "Adjusted Success Probability":
+                y_data.append(adjusted_success_prob)
+                y_data_unc.append(adjusted_success_prob_unc)
+
+            # Compute effective polarization and standard deviation (arXiv:2112.09853v1)
+            pol_factor = 4**self._num_qubits
+            pol = pol_factor / (pol_factor - 1) * adjusted_success_prob - 1 / (pol_factor - 1)
+            pol_unc = np.sqrt(pol_factor / (pol_factor - 1)) * adjusted_success_prob_unc
+            circ_result["metadata"]["polarization"] = pol
+            circ_result["metadata"]["polarization_uncertainty"] = pol_unc
+            if self.options.y_axis == "Effective Polarization":
+                y_data.append(pol)
+                y_data_unc.append(pol_unc)
+
+        return unp.uarray(y_data, y_data_unc)
+
+    def _initialize(
```

Comment:
This cannot be an `RBAnalysis` subclass? This method seems to me like almost a copy of that class.

Reply:
You're right, changed it to a subclass.